src/liballoc/heap.rs

```diff
@@ -44,6 +45,84 @@ fn check_size_and_alignment(size: usize, align: usize) {
                   align);
 }
 
+#[derive(Copy, Clone, Default, Debug)]
+pub struct HeapAllocator;
+
+unsafe impl Allocator for HeapAllocator {
+    unsafe fn alloc(&mut self, layout: Layout) -> Result<*mut u8, AllocErr> {
```

Comment:
Should this check for zero-sized `layout`s before passing to `alloc_unchecked`? The behavior is currently undefined on zero sized inputs.